diff --git a/video/Kbuild b/video/Kbuild
index aac7f52..438dcad 100644
--- a/video/Kbuild
+++ b/video/Kbuild
@@ -10,6 +10,7 @@ endif
 
 ifeq ($(CONFIG_MSM_VIDC_PINEAPPLE), y)
 LINUXINCLUDE   += -I$(VIDEO_DRIVER_ABS_PATH)/platform/pineapple/inc \
+                  -I$(VIDEO_DRIVER_ABS_PATH)/platform/cliffs/inc \
                   -I$(VIDEO_DRIVER_ABS_PATH)/variant/iris33/inc
 endif
 
@@ -24,7 +25,8 @@ USERINCLUDE    += -I$(VIDEO_ROOT)/include/uapi/vidc/media \
 obj-m += video.o
 
 ifeq ($(CONFIG_MSM_VIDC_PINEAPPLE), y)
-video-objs +=     $(VIDEO_DRIVER_REL_PATH)/platform/pineapple/src/pineapple.o \
+video-objs +=     $(VIDEO_DRIVER_REL_PATH)/platform/cliffs/src/cliffs.o \
+                  $(VIDEO_DRIVER_REL_PATH)/platform/pineapple/src/pineapple.o \
                   $(VIDEO_DRIVER_REL_PATH)/variant/iris33/src/msm_vidc_buffer_iris33.o \
                   $(VIDEO_DRIVER_REL_PATH)/variant/iris33/src/msm_vidc_power_iris33.o \
                   $(VIDEO_DRIVER_REL_PATH)/variant/iris33/src/msm_vidc_bus_iris33.o \
